src/core/abstract_types.jl

```diff
@@ -0,0 +1,10 @@
+# All subtypes of AbstractAffectFeedForward must define the field affected_variables.
+# TODO: make a unit test that checks for this.
+abstract type AbstractAffectFeedForward end
+
+abstract type AbstractCache end
+abstract type FeedForwardChronology end
+abstract type IniCondChronology end
```

Comment:
What does `IniCondChronology` stand for? `InitialConditionChronology`? Everything else is spelled out and I would prefer this to be too.